### ISSUE TITLE
Network name auto restore

### ISSUE THREAD

**Issue by FQrabbit** [Mygod/VPNHotspot#31]
Hi~

[Wireless Relay] When trying to change [Network Name], the wireless relay automatically turns off and reopens, the network name restores the default name.

`Device ：Google Nexus 6P  Android: 8.1.0`

[vpnhotspot-2701422247909376597.log](https://github.com/Mygod/VPNHotspot/files/2237573/vpnhotspot-2701422247909376597.log)

**Comment by Mygod** [comment]
You might be using an outdated `wpa_supplicant` driver since this issue is usually seen on Android 5.x devices. Are you using Google's official ROM?

Also, an Xposed module with package name starting with `com.chelpus` seems to be messing with your system's normal functionalities. Try disabling Xposed to see if that helps.

**Comment by FQrabbit** [comment]
![photo_2018-07-28_10-38-54](https://user-images.githubusercontent.com/29931248/43352213-ca4cbe74-9252-11e8-901a-9eddf926ce8c.jpg)
I am using Lineage OS,
I try to close Xposed
But I guess it may be a driver problem.

**Comment by FQrabbit** [comment]
[vpnhotspot-2532532893296176957.log](https://github.com/Mygod/VPNHotspot/files/2237631/vpnhotspot-2532532893296176957.log)
I tried to turn off Xposed. After restarting the phone, it is still the same problem. Maybe it is really a problem that has been driven for too long. :-1:

**Comment by Mygod** [comment]
Yeah at this point I'm afraid your phone doesn't support specifying repeater credentials. I'm using Lineage OS 15.1 on my other device and it's working well. Maybe you could try other ROMs?

**Comment by FQrabbit** [comment]
OK, I'll try to replace the ROM, thank you for your answer.

**Comment by Mygod** [comment]
I read a bit about VNDK and it seems like your problem might be resolved by replacing `[/system]/vendor/bin/hw/wpa_supplicant`. Could you try that instead?

**Comment by Mygod** [comment]
Also since you are using Lineage OS on an open source device, maybe you can ask your ROM maintainer to update `wpa_supplicant` driver (`platform/external/wpa_supplicant_8`).

**Comment by Mygod** [comment]
Hmm but if you're using Project Treble, system updates won't do anything because `wpa_supplicant` is placed in `/vendor`. Maybe check for `/vendor` updates?

**Comment by FQrabbit** [comment]
I am looking for the maintainer of the ROM. 🌝

**Comment by Mygod** [comment]
I found a device that I can reproduce this issue on. It seems like a new network is generated in `p2p_supplicant.conf` every time after `wpa_supplicant` is started and P2P GO is requested. I'll see what I can do about it.

**Comment by Mygod** [comment]
Okay. After some very intense debugging, I see this is a very complicated bug in AOSP that can only be triggered through `use_p2p_group_interface=1`. I'll open an issue later.

**Comment by Mygod** [comment]
Okay this patch to `external/wpa_supplicant_8` should hopefully fix the issue.

```c
diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index e9107b3b..32059352 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -7290,8 +7290,6 @@ static int nl80211_set_param(void *priv, const char *param)
        if (os_strstr(param, "use_p2p_group_interface=1")) {
                wpa_printf(MSG_DEBUG, "nl80211: Use separate P2P group "
                           "interface");
-               drv->capa.flags |= WPA_DRIVER_FLAGS_P2P_CONCURRENT;
-               drv->capa.flags |= WPA_DRIVER_FLAGS_P2P_MGMT_AND_NON_P2P;
        }
 #endif /* CONFIG_P2P */
 
diff --git a/src/drivers/driver_nl80211_capa.c b/src/drivers/driver_nl80211_capa.c
index 7064ce1d..f9404fe2 100644
--- a/src/drivers/driver_nl80211_capa.c
+++ b/src/drivers/driver_nl80211_capa.c
@@ -823,8 +823,6 @@ static int wpa_driver_nl80211_get_info(struct wpa_driver_nl80211_data *drv,
        if (info->p2p_concurrent) {
                wpa_printf(MSG_DEBUG, "nl80211: Use separate P2P group "
                           "interface (driver advertised support)");
-               drv->capa.flags |= WPA_DRIVER_FLAGS_P2P_CONCURRENT;
-               drv->capa.flags |= WPA_DRIVER_FLAGS_P2P_MGMT_AND_NON_P2P;
        }
        if (info->num_multichan_concurrent > 1) {
                wpa_printf(MSG_DEBUG, "nl80211: Enable multi-channel "
```

**Comment by snakwu** [comment]
> Okay this patch to `external/wpa_supplicant_8` should hopefully fix the issue.
> 
> ```c
> diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
> index e9107b3b..32059352 100644
> --- a/src/drivers/driver_nl80211.c
> +++ b/src/drivers/driver_nl80211.c
> @@ -7290,8 +7290,6 @@ static int nl80211_set_param(void *priv, const char *param)
>         if (os_strstr(param, "use_p2p_group_interface=1")) {
>                 wpa_printf(MSG_DEBUG, "nl80211: Use separate P2P group "
>                            "interface");
> -               drv->capa.flags |= WPA_DRIVER_FLAGS_P2P_CONCURRENT;
> -               drv->capa.flags |= WPA_DRIVER_FLAGS_P2P_MGMT_AND_NON_P2P;
>         }
>  #endif /* CONFIG_P2P */
>  
> diff --git a/src/drivers/driver_nl80211_capa.c b/src/drivers/driver_nl80211_capa.c
> index 7064ce1d..f9404fe2 100644
> --- a/src/drivers/driver_nl80211_capa.c
> +++ b/src/drivers/driver_nl80211_capa.c
> @@ -823,8 +823,6 @@ static int wpa_driver_nl80211_get_info(struct wpa_driver_nl80211_data *drv,
>         if (info->p2p_concurrent) {
>                 wpa_printf(MSG_DEBUG, "nl80211: Use separate P2P group "
>                            "interface (driver advertised support)");
> -               drv->capa.flags |= WPA_DRIVER_FLAGS_P2P_CONCURRENT;
> -               drv->capa.flags |= WPA_DRIVER_FLAGS_P2P_MGMT_AND_NON_P2P;
>         }
>         if (info->num_multichan_concurrent > 1) {
>                 wpa_printf(MSG_DEBUG, "nl80211: Enable multi-channel "
> ```

这问这个怎么用?我的也是三星手机,更改不了网络名和密码

**Comment by Mygod** [comment]
Upon further inspection, this bug can occur from Android 5 to 9. In fact, take darcy for example, this bug does not occur in Android 7.0 but is present on Android 8.0 (both are official ROMs). I will have to conclude that this is a ROM/vendor-specific bug and suggest everyone that encounters this bug to contact their vendors and see what they can do about this.

**Comment by Mygod** [comment]
Huh, I am surprised and delighted to find out that [Google has fixed this in Android Q](https://developer.android.com/reference/android/net/wifi/p2p/WifiP2pManager.html#createGroup(android.net.wifi.p2p.WifiP2pManager.Channel,%20android.net.wifi.p2p.WifiP2pConfig,%20android.net.wifi.p2p.WifiP2pManager.ActionListener)).

**Comment by Mygod** [comment]
Fixed via #90.
